src/main/java/DukeException.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * DukeException class is a custom exception class to handle
+ * problems that might occur in Duke.java
+ */
+public class DukeException extends Exception {
+
+    public DukeException(String errorMessage) {
+        super(errorMessage);
+    }
+
+    public DukeException() {}
+
+    /**
+     * checker method is used to run checks for all the different types of request
+     * and check if it invalids any of them. Returns a description or advice on what
+     * is missing or how to resolve it.
+     * @param arr This String array stores the user's input, used to check for validity
+     * @param curr Holds the capacity for the stored list, used to check if request is out of bounds
+     * @throws DukeException
+     */
+    public void checker(String[] arr, int curr) throws DukeException {
```

Comment:
maybe can use a more descriptive name for the method 😄 
e.g
commandChecker
stringChecker
inputTester